libyara/modules/macho.c

```diff
@@ -0,0 +1,1180 @@
+  set_integer(S_16BYTE_LITERALS, object, "S_16BYTE_LITERALS");
+  set_integer(S_DTRACE_DOF, object, "S_DTRACE_DOF");
+  set_integer(S_LAZY_DYLIB_SYMBOL_POINTERS, object,
+              "S_LAZY_DYLIB_SYMBOL_POINTERS");
+  set_integer(S_THREAD_LOCAL_REGULAR, object, "S_THREAD_LOCAL_REGULAR");
+  set_integer(S_THREAD_LOCAL_ZEROFILL, object, "S_THREAD_LOCAL_ZEROFILL");
+  set_integer(S_THREAD_LOCAL_VARIABLES, object, "S_THREAD_LOCAL_VARIABLES");
+  set_integer(S_THREAD_LOCAL_VARIABLE_POINTERS, object,
+              "S_THREAD_LOCAL_VARIABLE_POINTERS");
+  set_integer(S_THREAD_LOCAL_INIT_FUNCTION_POINTERS, object,
+              "S_THREAD_LOCAL_INIT_FUNCTION_POINTERS");
+
+  // Section attributes
+
+  set_integer(S_ATTR_PURE_INSTRUCTIONS, object, "S_ATTR_PURE_INSTRUCTIONS");
+  set_integer(S_ATTR_NO_TOC, object, "S_ATTR_NO_TOC");
+  set_integer(S_ATTR_STRIP_STATIC_SYMS, object, "S_ATTR_STRIP_STATIC_SYMS");
+  set_integer(S_ATTR_NO_DEAD_STRIP, object, "S_ATTR_NO_DEAD_STRIP");
+  set_integer(S_ATTR_LIVE_SUPPORT, object, "S_ATTR_LIVE_SUPPORT");
+  set_integer(S_ATTR_SELF_MODIFYING_CODE, object,
+              "S_ATTR_SELF_MODIFYING_CODE");
+  set_integer(S_ATTR_DEBUG, object, "S_ATTR_DEBUG");
+  set_integer(S_ATTR_SOME_INSTRUCTIONS, object, "S_ATTR_SOME_INSTRUCTIONS");
+  set_integer(S_ATTR_EXT_RELOC, object, "S_ATTR_EXT_RELOC");
+  set_integer(S_ATTR_LOC_RELOC, object, "S_ATTR_LOC_RELOC");
+}
+
+
+// Get Mach-O file index in fat file by cputype field.
+
+define_function(file_index_type)
+{
+  YR_OBJECT* module = module();
+
+  int64_t type_arg = integer_argument(1);
+
+  int64_t nfat = get_integer(module, "nfat_arch");
+  if (is_undefined(module, "nfat_arch"))
+    return_integer(UNDEFINED);
+
+  for (int64_t i = 0; i < nfat; ++i)
+  {
+    int64_t type = get_integer(module, "file[%i].cputype", i);
+    if (type == type_arg)
+    {
+      return_integer(i);
+    }
+  }
+  return_integer(UNDEFINED);
+}
+
+
+// Get Mach-O file index in fat file by cputype and cpusubtype fields.
+
+define_function(file_index_subtype)
+{
+  YR_OBJECT* module = module();
+
+  int64_t type_arg = integer_argument(1);
+  int64_t subtype_arg = integer_argument(2);
+
+  uint64_t nfat = get_integer(module, "nfat_arch");
+  if (is_undefined(module, "nfat_arch"))
+    return_integer(UNDEFINED);
+
+  for (uint64_t i = 0; i < nfat; ++i)
+  {
+    int64_t type = get_integer(module, "file[%i].cputype", i);
+    int64_t subtype = get_integer(module, "file[%i].cpusubtype", i);
+
+    if (type == type_arg && subtype == subtype_arg)
+    {
+      return_integer(i);
+    }
+  }
+  return_integer(UNDEFINED);
+}
+
+
+// Get real entry point offset for specific architecture in fat Mach-O.
+
+define_function(entry_point_for_arch_type)
+{
+  YR_OBJECT* module = module();
+
+  int64_t type_arg = integer_argument(1);
+
+  uint64_t nfat = get_integer(module, "nfat_arch");
+  if (is_undefined(module, "nfat_arch"))
+    return_integer(UNDEFINED);
+
+  for (uint64_t i = 0; i < nfat; ++i)
```

Comment:
no need to use a 64bit iterator here, the fat counter is ut32, also is better to use i++, because preincrement doesn't gives any benefit here